setup.py

```diff
@@ -132,7 +149,7 @@ def build_extension(self, ext):
                     ext.extra_compile_args.append("-isystem" + pyarrow.get_include())
                     ext.extra_compile_args.append("-isystem" + numpy.get_include())
                     if "std=" not in os.environ.get("CXXFLAGS", ""):
-                        ext.extra_compile_args.append("-std=c++11")
+                        ext.extra_compile_args.append("-std=c++17")
```

Comment:
@xhochy added this originally, I'd like him to take a look too

Reply:
here are the documentation pages stating the requirement for for C++17
https://arrow.apache.org/docs/developers/cpp/building.html#system-setup
https://arrow.apache.org/docs/cpp/conventions.html#language-version

Reply:
I mostly added the `if` to not override the `std` if it is passed in via `CXXFLAGS`. As we're keeping that, I'm happy. The change of `std` for the new Arrow is also correct.